src/transformers/file_utils.py

```diff
@@ -1242,8 +1242,15 @@ def get_from_cache(
             # between the HEAD and the GET (unlikely, but hey).
             if 300 <= r.status_code <= 399:
                 url_to_download = r.headers["Location"]
-        except (requests.exceptions.ConnectionError, requests.exceptions.Timeout):
-            # etag is already None
+        except (
+            requests.exceptions.ConnectionError,
+            requests.exceptions.Timeout,
+        ) as exc:
+            # Actually raise for those subclasses of ConnectionError:
+            if isinstance(exc, requests.exceptions.SSLError) or isinstance(exc, requests.exceptions.ProxyError):
+                raise exc
+            # Otherwise, our Internet connection is down.
+            # etag is None
```

Comment:
Not really the key point of the discussion, but it would be much more readable to write this:

- by putting just what can raise exceptions in the try/except block
- without isinstance checks -- except does that already

```python
        try:
            r = requests.head(url, headers=headers, allow_redirects=False, proxies=proxies, timeout=etag_timeout)
            r.raise_for_status()
        except (requests.exceptions.SSLError, requests.exceptions.ProxyError):
            # non recoverable errors -- user has to fix their system
            raise
        except (requests.exceptions.ConnectionError, requests.exceptions.Timeout):
            # recoverable errors -- likely the network failed during the download
            etag = None
        else:
            etag = r.headers.get("X-Linked-Etag") or r.headers.get("ETag")
            # rest of etag-related logic goes here
```